tests/code/test_events.py

```diff
@@ -519,33 +522,92 @@ def test_getAllFacilitators():
-    events = getUpcomingEventsForUser(user)
-    assert len(events) == 0
+def test_upcomingEvents():
+    with mainDB.atomic() as transaction:
+        testDate = datetime.datetime.strptime("2021-08-01 05:00","%Y-%m-%d %H:%M")
+
+        # Create a user to run the tests with
+        user = User.create(username = 'usrtst',
+                              firstName = 'Test',
+                              lastName = 'User',
+                              bnumber = '03522492',
+                              email = 'usert@berea.deu',
+                              isStudent = True)
+
+        # Create an event that is not a part of a program the user can RSVP to
+        noProgram = Event.create(name = "Upcoming event with no program",
+                                term = 2,
+                                description = "Test upcoming no program event.",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "The moon",
+                                startDate = "2021-12-12",
+                                endDate = "2021-12-13")
+
+        # Create a Program Event to show up when the user marks interest in a
+        # new program
+        newProgramEvent = Event.create(name = "Upcoming event with  program",
+                                term = 2,
+                                description = "Test upcoming program event.",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "The sun",
+                                startDate = 2021-12-12,
+                                endDate = 2021-12-13)
+
+        # Create a new Program to create the new Program Event off of so the
+        # user can mark interest for it
+        ProgramForInterest = Program.create(id = 13,
+                                            programName = "BOO",
+                                            isStudentLed = False,
+                                            isBonnerScholars = False,
+                                            emailReplyTo = "test@email",
+                                            emailSenderName = "testName")
+
+        ProgramEventForInterest = ProgramEvent.create(program = 13, event = newProgramEvent)
+
+        program_id = 13
+
+        # User has not RSVPd and is Interested
+        addInterest = addUserInterest(program_id, user)
+        eventsInUserInterestedProgram = getUpcomingEventsForUser(user, asOf = testDate)
+
+        eventsInProgram = list((Event.select()
+                                     .join(ProgramEvent)
+                                     .join(Program)
+                                     .where(Event.startDate >= testDate,
+                                            ProgramEvent.event_id == Event.id,
+                                            ProgramEvent.program_id == 13)))
```

Comment:
You created the events here for a new program, so you shouldn't need to do a big select like this